### PR TITLE
Add dockerfile

### PR DESCRIPTION
**Pull request by kaibocai**
Add dockerfile

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -15,6 +15,7 @@ import (
 
 var (
 	port       = flag.Int("port", 4001, "The server port")
+	ip         = flag.String("ip", "localhost", "The server IP")
```
Comment (cgillum) [inline main.go:18]: I think it's more common to use `host` as the name of the hostname parameter.

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+WORKDIR /root
+
+# Build
+RUN CGO_ENABLED=0 GOOS=linux go build -o /durabletask-go
+
+EXPOSE 4001
+
+# Run
+ENTRYPOINT [ "/durabletask-go" ]
+CMD [ "--ip", "0.0.0.0" ]
```
Comment (cgillum) [inline Dockerfile:15]: Should we make the port number explicit here too since we're hardcoding `EXPOSE 4001`?

File: main.go [hunk 3]
```diff
@@ -15,6 +15,7 @@ import (
 
 var (
 	port       = flag.Int("port", 4001, "The server port")
+	host       = flag.String("host", "localhost", "The server host")
```
Comment (ItalyPaleAle) [inline main.go:18]: Nit:

```suggestion
	host       = flag.String("host", "localhost", "The host to bind to")
```

File: Dockerfile [hunk 4]
```diff
@@ -0,0 +1,15 @@
+# syntax=docker/dockerfile:1
+
+FROM golang:1.21
```
Comment (ItalyPaleAle) [inline Dockerfile:3]: Please use a multi-stage Dockerfile:

1. Stage one, from “golang:1.21-alpine” just builds the container
2. Stage 2 is the production stage and it’s based on distroless.

Example: https://github.com/GoogleContainerTools/distroless/blob/main/examples/go/Dockerfile

**Review by ItalyPaleAle — CHANGES_REQUESTED** [review]

File: Dockerfile (hunk 4 above)
Comment (kaibocai) [reply Dockerfile]: thanks for this! Updated!

File: Dockerfile [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+
+FROM golang:1.21 as build
+
+COPY . /root
+WORKDIR /root
+
+# Build
+RUN CGO_ENABLED=0 GOOS=linux go build -o /durabletask-go
+
+FROM gcr.io/distroless/static-debian11
```
Comment (ItalyPaleAle) [inline Dockerfile:11]:
```suggestion
FROM gcr.io/distroless/static-debian12:nonroot
```

Comment (ItalyPaleAle) [reply Dockerfile:11]: There's a newer version and also we should use the "nonroot" variant if possible

**Review by cgillum — APPROVED** [review]
I'm good with these changes once @ItalyPaleAle's comments are resolved.

File: Dockerfile (hunk 5 above)
Comment (kaibocai) [reply Dockerfile]: updated!

**Review by ItalyPaleAle — APPROVED** [review]

**Pull request merged**